website/src/docs/companion.md

```diff
@@ -481,6 +486,25 @@ app.use(companion.app({
 
 We have [a detailed guide on running Companion in Kubernetes](https://github.com/transloadit/uppy/blob/main/packages/%40uppy/companion/KUBERNETES.md) for you, that’s how we run our example server at <https://companion.uppy.io>.
 
+### Running many instances
+
+Two ways of running many concurrent Companion instances.
+
+#### Separate endpoints
+
+One option is to run many instances with each instance having its own unique endpoint. This could be on separate ports, (sub)domain names, or IPs. With this setup, the Companion client in Uppy will direct requests to companion in a way so that all requests of a particular upload will always be sent to the same endpoint, and hence process. You would then also typically configure a single instance (one endpoint) to handle all OAuth authentication requests, so that you only need to specify a single OAuth callback URL. See also `oauthDomain` and `validHosts`
```

Comment:
How does this work in Uppy where you can only set a single `companionUrl`?

Reply:
I think this code https://github.com/transloadit/uppy/blob/bffd5dc9ca1d945ac2c566ca0328d5a174cc4cce/packages/%40uppy/companion-client/src/RequestClient.js#L79

 and this https://github.com/transloadit/uppy/blob/bffd5dc9ca1d945ac2c566ca0328d5a174cc4cce/packages/%40uppy/companion-client/src/RequestClient.js#L57

I'll try to explain it

Reply:
side note: sending i-am from companion is enabled by setting COMPANION_SELF_ENDPOINT. However api2 does not set this env variable, so it does not use this functionality (any more).

However the Transloadit Uppy plugin has its own logic where it overrides companionUrl as soon as it receives a response from the created assembly:

https://github.com/transloadit/uppy/blob/5c0967fcdea7036517d0c480a6ff7b005e67d5b6/packages/%40uppy/transloadit/src/index.js#L163

So this is how Uppy/Transloadit pins each upload to a single companion instance.